model/project_parser.go

```diff
@@ -800,32 +796,45 @@ func GetProjectFromFile(ctx context.Context, opts GetProjectOpts) (ProjectInfo,
 // intermediate project representation (i.e. before selectors or
 // matrix logic has been evaluated).
 // If unmarshalStrict is true, use the strict version of unmarshalling.
-func createIntermediateProject(yml []byte, unmarshalStrict bool) (*ParserProject, error) {
+func createIntermediateProject(yml []byte, unmarshalStrict bool) (*ParserProject, *ProjectConfig, error) {
+	c := ProjectConfig{}
 	p := ParserProject{}
 	if unmarshalStrict {
 		strictProjectWithVariables := struct {
-			ParserProject `yaml:"pp,inline"`
+			ParserProject       `yaml:"pp,inline"`
+			ProjectConfigFields `yaml:"pc,inline"`
 			// Variables is only used to suppress yaml unmarshalling errors related
 			// to a non-existent variables field.
 			Variables interface{} `yaml:"variables,omitempty" bson:"-"`
 		}{}
 
 		if err := util.UnmarshalYAMLStrictWithFallback(yml, &strictProjectWithVariables); err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		p = strictProjectWithVariables.ParserProject
+		c = ProjectConfig{
+			"",
```

Comment:
We don't define things like this without specifying the field; can you just set ` c = ProjectConfig{ProjectConfigFields: strictProjectWithVariables.ProjectConfigFields}` ?